gi_wrapper_cpp/dicom_tag_strategies.cpp

```diff
@@ -66,69 +74,50 @@ bool omitSpecifiedTagsStrategy(int tagGroupId, int tagTagId, set <pair<int, int>
 // 1 or other integer values - rewrite all tags from source dicom
 tagsRewriteStrategy resolveTagsRewriteStrategy(int strategy)
 {
-    switch(strategy)
+    switch (strategy)
     {
-        case 5:
-            return omitSpecifiedTagsStrategy;
-        case 4:
-            return rewriteSpecifiedTagsStrategy;
-        case 3:
-            return rewriteAllNonPrivateTagsStrategy;
-        case 2:
-            return rewriteNoTagsStrategy;
-        default:
-            return rewriteAllTagsStrategy;
+    case 5:
+        return omitSpecifiedTagsStrategy;
+    case 4:
+        return rewriteSpecifiedTagsStrategy;
+    case 3:
+        return rewriteAllNonPrivateTagsStrategy;
+    case 2:
+        return rewriteNoTagsStrategy;
+    default:
+        return rewriteAllTagsStrategy;
     }
 }
 
-void applyDicomTagsRewriteStrategy(DataSet* source, DataSet* destination, int rewriteTagsStrategyId, set <pair<int, int>> userTags)
+void applyDicomTagsRewriteStrategy(DataSet& source, DataSet& destination, int rewriteTagsStrategyId, set <pair<int, int>> userTags)
 {
     // Map strategy passed by parameter from integer value to pointer to function represent strategy
     tagsRewriteStrategy rewriteStrategy = resolveTagsRewriteStrategy(rewriteTagsStrategyId);
 
-    // Get tags from source DICOM
-    tagsIds_t tags = source->getTags();
-
-    // Iterate over tags from source DICOM
-    for (tagsIds_t::const_iterator scanTags = tags.begin(), endTags = tags.end(); scanTags != endTags; ++scanTags)
-    {
-        // Using selected strategy check, if tag (scanTags->getGroupId(), scanTags->getTagId()) should be rewritten into result DICOM
-        if (rewriteStrategy(scanTags->getGroupId(), scanTags->getTagId(), userTags))
+    for (DataSet::ConstIterator des = destination.Begin(), desEnd = destination.End(); des != desEnd; ++des)
+    {   
+        if (rewriteStrategy(des->GetTag().GetGroup(), des->GetTag().GetElement(), userTags))
         {
             // 0x7fe0 = Pixel_Data tagId - we create new data so we shouldn't rewrite this tag from input dicom
-            if (scanTags->getGroupId() == 0x7fe0)
+            if (des->GetTag().GetGroup() == 0x7FE0) {
```

Comment:
@kdbogusz It would be nice to extract this 0x7FE0 constant to some named constant like:
constexpr int64_t <name-of-constant> = 0x7FE0;
This should apply to most of the constants + they all should be placed in another file (preferable header file)